pkg/network/namescheme/networknamescheme.go

```diff
@@ -94,3 +97,37 @@ func generateOrdinalInterfaceName(idx int) string {
 	const ordinalIfacePrefix = "net"
 	return fmt.Sprintf("%s%d", ordinalIfacePrefix, idx)
 }
+
+// CreateNetworkNameSchemeByPodNetworkStatus create pod network name scheme according to the given VMI spec networks
+// and pod network status.
+// In case the pod network status has at least one interface with ordinal interface name it returns the ordinal network
+// name-scheme, Otherwise, returns the hashed network name scheme.
+func CreateNetworkNameSchemeByPodNetworkStatus(networks []v1.Network, networkStatus map[string]networkv1.NetworkStatus) map[string]string {
```

Comment:
> It feels to me that the safest is to make `CreateOrdinalNetworkNameScheme`and `CreateNetworkNameScheme` non exported. And export only this new method. 

This method can be used only where the pod Multus network-status annotation is in hand - virt-controller.
Any other place that need to use both name schemes cannot use this one, and should use other factories instead.

> I don't understand how for example [sriov](https://github.com/kubevirt/kubevirt/blob/main/pkg/network/sriov/sriov.go#L64) and [configState](https://github.com/kubevirt/kubevirt/blob/main/pkg/network/setup/configstate.go#L55) will work for old VMs. It seems they are using the new name scheme only.

Regarding the SR-IOV, its not covered my bad, I will fix it.

As for the configState, I had no idea such a change is going on.
It basically breaks this PR because it assumes the network setup enters the pod network namespace much earlier, and the correct pod interface name was found.

But now we cannot get the correct pod iface name at the right time, causing the configState to look for the wrong cache file, and perform network setup again.
It fails with a critical error and eventually, the VM dies.

> Did you run upgrade tests? If so, I don't understand how they passed. Per my understanding, virt-hanlder will try to config the network for all the interfaces although they are already configured and will get a critical failure.

AFAIK upgrade tests  are part of sig-operator lane and runs on any PR.
Per my understanding, virt-handler won't enter network-setup for running VMs unless the spec and status interfaces are not aligned (e.g.: hotplug iface).